Added auth logic

Marcus-2k opened the pull request.

Marcus-2k committed:
added auth logic

Puchkof commented on ASP-NET/ASP-NET/Controllers/Auth.cs:
```diff
@@ -0,0 +1,113 @@
+        [HttpPost("login")]
+        public async Task<ActionResult<ResultResponseDto>> Login(LoginDto request)
+        {
+            Console.WriteLine(request.Email);
```
Use ILogger<> instead

Puchkof commented on ASP-NET/ASP-NET/Controllers/Auth.cs:
```diff
@@ -0,0 +1,113 @@
+        {
+            Console.WriteLine(request.Email);
+
+            var user = await this._userManager.FindByEmailAsync(request.Email);
```
Please follow N-layer architecture principles and move all the business logic to application layer(create services) in all the controllers

Marcus-2k committed:
update auth logic

1. rename file "auth" to "AuthController"
2. Removed BCrypt package
3. Check if a user with this name exists

Marcus-2k committed:
fix bag with bracket duplication

Puchkof commented on ASP-NET/ASP-NET/Context/ApplicationDbContext.cs:
```diff
@@ -35,7 +37,13 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 .HasForeignKey(pc => pc.CategoryId);
 
-            base.OnModelCreating(modelBuilder);
+            modelBuilder.Entity<User>(entity => entity.ToTable("Users"));
```
I am not sure if you need to rename these table, I would use default names

Puchkof commented on ASP-NET/ASP-NET/Controllers/AuthController.cs:
```diff
@@ -0,0 +1,124 @@
+
+            var authClaims = new[]
+                {
+                    new Claim("Username", user.UserName),
```
I would also add user id to token to reuse it in db queries in future

Puchkof commented on ASP-NET/ASP-NET/Controllers/AuthController.cs:
```diff
@@ -0,0 +1,124 @@
+                Value = token
+            };
+
+            _context.Tokens.Add(tokenEntity);
```
You dont need to store token in db

Marcus-2k committed:
transferring logic to the controller service
